packages/dcos-integration-test/extra/test_units.py

```diff
@@ -66,3 +70,58 @@ def _check_line(line):
 
     _check_units("/etc/systemd/system/dcos-*.service")
     _check_units("/etc/systemd/system/dcos-*.socket")
+
+
+def test_socket_units():
+    """Test that socket units configure socket files in /run/dcos
+    that are owned by 'dcos_adminrouter'.
+    """
+    def _check_unit(file):
+        logging.info("Checking socket unit {}".format(file))
+        out = subprocess.check_output(
+            ["/usr/bin/systemctl", "show", "--no-pager", os.path.basename(file)],
+            stderr=subprocess.STDOUT,
+            universal_newlines=True)
+        user = ""
+        group = ""
+        mode = ""
+        had_unix_socket = False
+        for line in out.split("\n"):
+            parts = line.split("=")
+            if len(parts) != 2:
+                continue
+            k, v = parts
+            if k == "SocketUser":
+                user = v
+            if k == "SocketGroup":
+                group = v
+            if k == "ListenStream":
+                # Unix sockets are distinguished from IP sockets by having a '/' as the first
+                # character in the value of the ListenStream directive.
+                if v.startswith("/"):
+                    had_unix_socket = True
+                    assert v.startswith("/run/dcos/"), "DC/OS unix sockets must go in the /run/dcos directory"
+            if k == "SocketMode":
+                mode = v
+        if not had_unix_socket:
+            # This socket file doesn't declare any unix sockets, ignore.
+            return
+        assert user == "root"
+        assert group == "dcos_adminrouter"
+        assert mode == "0660"
+
+    for file in glob.glob("/etc/systemd/system/dcos-*.socket"):
```

Comment:
If someone wonders why `file` is orange here: the GitHub Python syntax highlighting is obviously still focusing on Python 2, where `file` was a built-in function. :-).